The gapis server has exitedd with an error code of: 2

panic: runtime error: invalid memory address or nil pointer dereference
  panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0xff93e7]

goroutine 12 [running]:
github.com/google/gapid/gapis/gfxapi/gles.compat.func4(0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e4bf20)
	/home/mao/go/src/github.com/google/gapid/gapis/gfxapi/gles/compat.go:1034 +0x8047
github.com/google/gapid/gapis/atom/transform.transform.Transform(0x1fcde76, 0x6, 0xc424938dc0, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e4bf20)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:75 +0x75
github.com/google/gapid/gapis/atom/transform.(*transform).Transform(0xc424389c40, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e4bf20)
	<autogenerated>:10 +0xb1
github.com/google/gapid/gapis/atom/transform.TransformWriter.MutateAndWrite(0xc4261fd680, 0x3265360, 0xc424389c40, 0x3265320, 0xc420e4bf20, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:98 +0x83
github.com/google/gapid/gapis/atom/transform.(*TransformWriter).MutateAndWrite(0xc420e4bf50, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040)
	<autogenerated>:8 +0xa1
github.com/google/gapid/gapis/gfxapi/gles.(*readFramebuffer).Transform(0xc425b5a570, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e4bf50)
	/home/mao/go/src/github.com/google/gapid/gapis/gfxapi/gles/read_framebuffer.go:46 +0x73
github.com/google/gapid/gapis/atom/transform.TransformWriter.MutateAndWrite(0xc4261fd680, 0x32631e0, 0xc425b5a570, 0x3265320, 0xc420e4bf50, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:98 +0x83
github.com/google/gapid/gapis/atom/transform.(*TransformWriter).MutateAndWrite(0xc420e04000, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040)
	<autogenerated>:8 +0xa1
github.com/google/gapid/gapis/gfxapi/gles.undefinedFramebuffer.func1(0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e04000)
	/home/mao/go/src/github.com/google/gapid/gapis/gfxapi/gles/undefined_framebuffer.go:32 +0x9c
github.com/google/gapid/gapis/atom/transform.transform.Transform(0x1fdbbde, 0x10, 0xc424389940, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e04000)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:75 +0x75
github.com/google/gapid/gapis/atom/transform.(*transform).Transform(0xc424389960, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e04000)
	<autogenerated>:10 +0xb1
github.com/google/gapid/gapis/atom/transform.TransformWriter.MutateAndWrite(0xc4261fd680, 0x3265360, 0xc424389960, 0x3265320, 0xc420e04000, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:98 +0x83
github.com/google/gapid/gapis/atom/transform.(*TransformWriter).MutateAndWrite(0xc420e04030, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040)
	<autogenerated>:8 +0xa1
github.com/google/gapid/gapis/gfxapi/gles.(*findIssues).Transform(0xc426641590, 0x3285160, 0xc4206e2c00, 0x94, 0x3294e20, 0xc42093c040, 0x3265320, 0xc420e04030)
	/home/mao/go/src/github.com/google/gapid/gapis/gfxapi/gles/find_issues.go:102 +0x168
github.com/google/gapid/gapis/atom/transform.TransformWriter.MutateAndWrite(0xc4261fd680, 0x32631a0, 0xc426641590, 0x3265320, 0xc420e04030, 0x3285160, 0xc420e4bad0, 0x94, 0x3294e20, 0xc42093c040)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:98 +0x83
github.com/google/gapid/gapis/atom/transform.(*TransformWriter).MutateAndWrite(0xc420e04090, 0x3285160, 0xc420e4bad0, 0x94, 0x3294e20, 0xc42093c040)
	<autogenerated>:8 +0xa1
github.com/google/gapid/gapis/atom/transform.Transforms.Transform(0xc424d2c580, 0x5, 0x8, 0x3285160, 0xc420e4bad0, 0xc423dda000, 0x18a9, 0x1e00, 0x3263460, 0xc4209578a0)
	/home/mao/go/src/github.com/google/gapid/gapis/atom/transform/transforms.go:40 +0x1ba
github.com/google/gapid/gapis/gfxapi/gles.api.Replay(0x3285160, 0xc420e4b830, 0xc425b5a450, 0xc425b5a498, 0x1c91f40, 0x3bddeb8, 0xc424389160, 0x1, 0x1, 0x32f7160, ...)
	/home/mao/go/src/github.com/google/gapid/gapis/gfxapi/gles/replay.go:205 +0xb80
github.com/google/gapid/gapis/gfxapi/gles.(*api).Replay(0x3bddeb8, 0x3285160, 0xc420e4b830, 0xc425b5a450, 0xc425b5a498, 0x1c91f40, 0x3bddeb8, 0xc424389160, 0x1, 0x1, ...)
	<autogenerated>:3461 +0xfb
github.com/google/gapid/gapis/replay.(*Manager).execute(0xc420370460, 0x3285160, 0xc420e4b830, 0x6fe99ac9d6aaba76, 0x5b91f35ea5fec414, 0x52571307bb54a471, 0x97a5609c9ce21117, 0xb8e2683d7a19e62b, 0x1c91f40, 0x3bddeb8, ...)
	/home/mao/go/src/github.com/google/gapid/gapis/replay/batch.go:135 +0xa04
github.com/google/gapid/gapis/replay.(*Manager).batch(0xc420370460, 0x3285160, 0xc420e4b620, 0xc424388b60, 0x1, 0x1, 0x1d4f4c0, 0xc4206c1020, 0x1dfda40, 0xc4206c8780, ...)
	/home/mao/go/src/github.com/google/gapid/gapis/replay/batch.go:68 +0x514
github.com/google/gapid/gapis/replay.(*Manager).(github.com/google/gapid/gapis/replay.batch)-fm(0x3285160, 0xc420317d10, 0xc424388b60, 0x1, 0x1, 0x1d4f4c0, 0xc4206c1020, 0x1dfda40, 0xc4206c8780, 0x0)
	/home/mao/go/src/github.com/google/gapid/gapis/replay/manager.go:126 +0x93
github.com/google/gapid/gapis/replay/scheduler.(*bin).exec(0xc423f78e00, 0x3285160, 0xc420317d10, 0xc42024bf60)
	/home/mao/go/src/github.com/google/gapid/gapis/replay/scheduler/scheduler.go:241 +0x2d8
github.com/google/gapid/gapis/replay/scheduler.(*Scheduler).run(0xc420371ba0, 0x3285160, 0xc420317d10)
	/home/mao/go/src/github.com/google/gapid/gapis/replay/scheduler/scheduler.go:174 +0x469
created by github.com/google/gapid/gapis/replay/scheduler.New
	/home/mao/go/src/github.com/google/gapid/gapis/replay/scheduler/scheduler.go:72 +0xd0

--- a/gapis/gfxapi/gles/find_issues.go
+++ b/gapis/gfxapi/gles/find_issues.go
@@ -91,14 +91,37 @@ func (e ErrUnexpectedDriverTraceError) Error() string {
 func (t *findIssues) Transform(ctx context.Context, i atom.ID, a atom.Atom, out transform.Writer) {
 	ctx = log.Enter(ctx, "findIssues")
 	t.lastGlError = GLenum_GL_NO_ERROR
-	if err := a.Mutate(ctx, t.state, nil /* no builder */); err != nil {
-		if atom.IsAbortedError(err) && t.lastGlError != GLenum_GL_NO_ERROR {
+	mutateErr := a.Mutate(ctx, t.state, nil /* no builder */)
+	if mutateErr != nil {
+		if atom.IsAbortedError(mutateErr) && t.lastGlError != GLenum_GL_NO_ERROR {
 			// GL errors have already been reported - so do not log it again.
 		} else {
-			t.onIssue(a, i, service.Severity_ErrorLevel, err)
+			t.onIssue(a, i, service.Severity_ErrorLevel, mutateErr)
 		}
 	}
+
 	mutatorsGlError := t.lastGlError
+	if e := FindErrorState(a.Extras()); e != nil {
+		// Check that our API file agrees with the driver which we used for tracking.
+		if (e.TraceDriversGlError != GLenum_GL_NO_ERROR) != (mutatorsGlError != GLenum_GL_NO_ERROR) {
+			errorMsg := ErrUnexpectedDriverTraceError{
+				DriverError:   e.TraceDriversGlError,
+				ExpectedError: mutatorsGlError,
+			}
+			t.onIssue(a, i, service.Severity_FatalLevel, errorMsg)
+		}
+		// Check that the C++ and Go versions of the generated code precisely agree.
+		if e.InterceptorsGlError != mutatorsGlError {
+			t.onIssue(a, i, service.Severity_FatalLevel, fmt.Errorf("%s in interceptor, but we expected %s",
+				e.InterceptorsGlError.ErrorString(), mutatorsGlError.ErrorString()))
+		}
+	}
+
+	if mutateErr != nil {
+		// Ignore since downstream transform layers can only consume valid commands
+		return
+	}
+
 	out.MutateAndWrite(ctx, i, a)
 
 	dID := i.Derived()
@@ -117,32 +140,14 @@ func (t *findIssues) Transform(ctx context.Context, i atom.ID, a atom.Atom, out 
 			if err != nil {
 				return err
 			}
-			v := r.Uint32()
+			v := GLenum(r.Uint32())
 			err = r.Error()
 			if err != nil {
 				t.onIssue(a, i, service.Severity_FatalLevel, fmt.Errorf("Failed to decode glGetError postback: %v", err))
 				return err
 			}
-			replayDriversGlError := GLenum(v)
-			if e := FindErrorState(a.Extras()); e != nil {
-				// Check that our API file agrees with the driver which we used for tracking.
-				if (e.TraceDriversGlError != GLenum_GL_NO_ERROR) != (mutatorsGlError != GLenum_GL_NO_ERROR) {
-					err := ErrUnexpectedDriverTraceError{
-						DriverError:   e.TraceDriversGlError,
-						ExpectedError: mutatorsGlError,
-					}
-					t.onIssue(a, i, service.Severity_FatalLevel, err)
-				}
-				// Check that the C++ and Go versions of the generated code precisely agree.
-				if e.InterceptorsGlError != mutatorsGlError {
-					t.onIssue(a, i, service.Severity_FatalLevel, fmt.Errorf("%s in interceptor, but we expected %s",
-						e.InterceptorsGlError.ErrorString(), mutatorsGlError.ErrorString()))
-				}
-			}
-			// The compatibility layer should not produce errors but it can propagate or silence them.
-			if (replayDriversGlError != GLenum_GL_NO_ERROR) && (mutatorsGlError == GLenum_GL_NO_ERROR) {
-				t.onIssue(a, i, service.Severity_FatalLevel, fmt.Errorf("%s in replay driver, but we expected %s",
-					replayDriversGlError.ErrorString(), mutatorsGlError.ErrorString()))
+			if v != GLenum_GL_NO_ERROR {
+				t.onIssue(a, i, service.Severity_FatalLevel, fmt.Errorf("%v in replay driver", v))
 			}
 			return nil
 		}))
@@ -158,12 +163,6 @@ func (t *findIssues) Transform(ctx context.Context, i atom.ID, a atom.Atom, out 
 			}
 		}
 		return strings.TrimSpace(s)
-	}
-
-	// Do not do any further checks if we have already determined the command fails.
-	// Furthermore, invalid commands run into risk of accessing invalid state.
-	if mutatorsGlError != GLenum_GL_NO_ERROR {
-		return
 	}
 
 	switch a := a.(type) {

